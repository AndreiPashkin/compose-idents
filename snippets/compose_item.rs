use compose_idents::compose_item;

#[compose_item(
    my_fn = concat(foo, _, bar),
)]
pub fn my_fn() -> u32 {
    42
}

fn main() {
    assert_eq!(foo_bar(), 42);
}
