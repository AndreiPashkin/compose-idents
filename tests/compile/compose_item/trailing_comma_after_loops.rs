//! A trailing comma after loops (with no alias spec) should be rejected for `#[compose_item]`.
use compose_idents::compose_item;

#[compose_item(
    for name in [foo, bar],
)]
fn name() {}

fn main() {}
