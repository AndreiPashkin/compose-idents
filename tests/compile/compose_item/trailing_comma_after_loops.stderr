error: Trailing separator after loops is not allowed in #[compose_item].
 --> tests/compile/compose_item/trailing_comma_after_loops.rs:5:27
  |
5 |     for name in [foo, bar],
  |                           ^
